clients/mobile/src/client.rs

```diff
@@ -0,0 +1,54 @@
+use crate::dto::ErrorDTO;
+use crate::{dto::TodoListDTO, runtime::Runtime};
+use application::command::Command;
+use application::query::GetTodoListQuery;
+use framework::*;
+use std::sync::Arc;
+
+#[derive(uniffi::Object)]
+pub struct Client {
+    runtime: Runtime,
+}
+
+#[uniffi::export]
+impl Client {
+    #[uniffi::constructor]
+    pub fn new() -> Arc<Self> {
+        Arc::new(Self {
+            runtime: Runtime::default(),
+        })
+    }
+
+    pub async fn add_task(&self, name: String) -> Result<(), ErrorDTO> {
+        Command::AddTask { name }
+            .execute(&self.runtime)
+            .await
+            .map_err(ErrorDTO::from)
+    }
+
+    pub async fn remove_task(&self, index: u32) -> Result<(), ErrorDTO> {
+        Command::RemoveTask {
+            index: index as usize,
+        }
+        .execute(&self.runtime)
+        .await
+        .map_err(ErrorDTO::from)
+    }
+
+    pub async fn complete_task(&self, index: u32) -> Result<(), ErrorDTO> {
+        Command::CompleteTask {
+            index: index as usize,
+        }
+        .execute(&self.runtime)
+        .await
+        .map_err(ErrorDTO::from)
+    }
+
+    pub async fn get_todo_list(&self) -> Result<TodoListDTO, ErrorDTO> {
+        GetTodoListQuery {}
+            .execute(&self.runtime)
+            .await
+            .map_err(ErrorDTO::from)
+            .map(TodoListDTO::from)
+    }
```

Comment:
**Suggestion:** Consider using more descriptive function names like `add_new_task`, `remove_existing_task`, `mark_task_as_completed` and `fetch_todo_list`. [enhancement]
```suggestion
    pub async fn add_new_task(&self, name: String) -> Result<(), ErrorDTO> { ... }
    pub async fn remove_existing_task(&self, index: u32) -> Result<(), ErrorDTO> { ... }
    pub async fn mark_task_as_completed(&self, index: u32) -> Result<(), ErrorDTO> { ... }
    pub async fn fetch_todo_list(&self) -> Result<TodoListDTO, ErrorDTO> { ... }
```